src/dialogs/permissions.jsx

```diff
@@ -115,6 +164,16 @@ const EditPermissionsModal = ({ dialogResult, selected, path }) => {
         }
     };
 
+    const spawnEncloseFiles = async () => {
+        try {
+            await cockpit.spawn(["chmod", "-R", mode_to_args(mode), path],
```

Comment:
Because of using only `path` this changes permissions on all enclosed files in `cwd` rather than in the selected directory.

Reply:
Oh you are right, this code is confusing! And I thought I could optimize it....